### PR TITLE
feat(PLM-49): Support RegEx - Default Search Method

### PR DESCRIPTION
**Pull request by kilikkuo**
Why need this change ? / Root Cause :
--------------

Changed Made :
--------------

Test Scope / Change Impact :
--------------

### CODE REVIEW

File: data_spec_validator/spec/validators.py [hunk 1]
```diff
@@ -370,3 +371,27 @@ def validate(value, extra, data):
...
+    name = REGEX
+
+    @staticmethod
+    def validate(value, extra, data):
+        regex_param = extra.get(RegexValidator.name, {})
+        pattern = regex_param.get('pattern', '')
+        match_method = regex_param.get('method', 'search')
+        error_regex_param = regex_param.copy()
+        if 'method' not in regex_param:
+            error_regex_param['method'] = 'search'
```
Comment (xeonchen) [inline data_spec_validator/spec/validators.py:386]: is `error_regex_param['method'] = match_method` better?

File: data_spec_validator/spec/validators.py [hunk 2]
```diff
@@ -370,3 +371,27 @@ def validate(value, extra, data):
...
+
+        if match_method == 'match':
+            match_func = re.match
+        elif match_method == 'fullmatch':
+            match_func = re.fullmatch
+        else:
+            match_func = re.search
```
Comment (xeonchen) [inline data_spec_validator/spec/validators.py:393]: I'd prefer

```
        if match_method == 'match':
            match_func = re.match
        elif match_method == 'fullmatch':
            match_func = re.fullmatch
        elif match_method == 'search':
            match_func = re.search
        else:
            assert False, 'unsupported match method'
```

Comment (kilikkuo) [reply data_spec_validator/spec/validators.py]: addressed, thanks

**Review by xeonchen — APPROVED** [review]

File: .git-blame-ignore-revs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+# Run "autoflake"
+
+# Run "isort"
+
+# Run "black"
+
+# Run pre-commit on missed files
+
+# Run "prettier"
```
Comment (xeonchen) [inline .git-blame-ignore-revs:14]: You don't really need these.

Comment (kilikkuo) [reply .git-blame-ignore-revs:14]: lol

Comment (kilikkuo) [reply .git-blame-ignore-revs]: addressed

File: data_spec_validator/spec/validators.py (hunk 1 above)
Comment (kilikkuo) [reply data_spec_validator/spec/validators.py]: Nice, addressed

**Pull request merged**